storageClient/api.go

```diff
@@ -113,6 +114,27 @@ type GetMembershipKeyRequest struct {
 	GroupID uuid.UUID `json:"group_id"`
 }
 
+// ListGroupRecordsRequest wraps values needed for the request to list all records shared with a group, can filter by writer ids
+type ListGroupRecordsRequest struct {
+	GroupID   uuid.UUID `json:"group_id"`
+	WriterIDs []string  `json:"writer_ids"`
+	NextToken int64
+	Max       int
+}
+
+// ListGroupRecordsResponse returns all the records shared with a group and the values needed to unwrap.
+type ListGroupRecordsResponse struct {
+	ResultList        []pdsClient.ListedRecord `json:"results"`
+	LastIndex         int                      `json:"last_index"`
```

Comment:
It honestly doesn’t matter which way you choose, but use either LastIndex OR (exclusive or ;-) NextToken for the request and response